### PR TITLE
Providing a PSR-7 Request instance

### PR DESCRIPTION
**Pull request by rtheunissen**
This is primarily a proof of concept to complement #315, which provides a way to create an authorized request, which the client or something else can send.

### CODE REVIEW

File: src/Provider/AbstractProvider.php [hunk 1]
```diff
@@ -260,31 +258,15 @@ public function getAccessToken($grant = 'authorization_code', array $params = []
...
-                // @codeCoverageIgnoreStart
-                default:
-                    throw new \InvalidArgumentException('Neither GET nor POST is specified for request');
-                // @codeCoverageIgnoreEnd
-            }
-        } catch (HttpAdapterException $e) {
-            $response = (string) $e->getResponse()->getBody();
+            $method = $this->method;
+
+            $response = $client->$method($this->urlAccessToken(), [
```
Comment (rtheunissen) [inline src/Provider/AbstractProvider.php]: Felt like cleaning this up a bit, removed the `switch`.

File: src/Provider/AbstractProvider.php [hunk 2]
```diff
@@ -295,24 +277,16 @@ public function getAccessToken($grant = 'authorization_code', array $params = []
...
-     * Creates a PSR-7 compatible request instance that can be modified.
-     * Often used to create calls against an API that requires authentication.
-     *
-     * @param  string $method
-     * @param  string $url
-     * @param  AccessToken $token
-     * @return RequestInterface
      */
-    public function getAuthenticatedRequest($method, $url, AccessToken $token)
+    public function getRequest($method, $url, AccessToken $token, $body = null)
```
Comment (rtheunissen) [inline src/Provider/AbstractProvider.php]: Figured a provider request might not be authorized if the access token is null, or if it's considered to always be authorized then why not call it `getRequest` to be consistent with `getResponse`.

**Comment by ramsey** [comment]
I don't understand how this differs from #315. Additionally, there are conflicts, so I can't merge it.

**Comment by rtheunissen** [comment]
Fair enough, would have made more sense to fork #315 instead.

**Pull request closed without merging**